components/cli/cmd/cellery/init.go

```diff
@@ -19,28 +19,47 @@
 package main
 
 import (
-	"github.com/spf13/cobra"
+	"fmt"
+	"strings"
 
 	"github.com/cellery-io/sdk/components/cli/cli"
 	"github.com/cellery-io/sdk/components/cli/pkg/commands/project"
 	"github.com/cellery-io/sdk/components/cli/pkg/util"
+	"github.com/spf13/cobra"
 )
 
 func newInitCommand(cli cli.Cli) *cobra.Command {
 	var projectName = ""
+	var testStr = ""
 	cmd := &cobra.Command{
 		Use:   "init [PROJECT_NAME]",
 		Short: "Initialize a cell project",
-		Args:  cobra.MaximumNArgs(1),
-		Run: func(cmd *cobra.Command, args []string) {
-			if len(args) > 0 {
-				projectName = args[0]
+		Args: func(cmd *cobra.Command, args []string) error {
+			err := cobra.MaximumNArgs(2)(cmd, args)
+			if err != nil {
+				return err
 			}
-			if err := project.RunInit(cli, projectName); err != nil {
-				util.ExitWithErrorMessage("Cellery init command failed", err)
+
+			if len(args) > 1 {
+				testStr = args[0]
+				if testStr != "test" {
+					return fmt.Errorf("invalid argument. expects %s, recieved %s", util.Bold("test"), testStr)
+				}
+				projectName = args[1]
+				if !(strings.HasSuffix(projectName, ".bal")) {
+					return fmt.Errorf("expects a bal file, recieved %v", projectName)
+				}
+			} else if len(args) > 0 {
+				testStr = ""
+				projectName = args[0]
 			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			project.RunInit(projectName, testStr)
```

Comment:
Handle the error return from RunInit function here.

Reply:
Fixed